charms/worker/k8s/src/charm.py

```diff
@@ -362,10 +371,29 @@ def _check_k8sd_ready(self):
 
     def _get_extra_sans(self):
         """Retrieve the certificate extra SANs."""
+        # Get the extra SANs from the configuration
         extra_sans_str = str(self.config.get("kube-apiserver-extra-sans") or "")
-        configured_sans = {san for san in extra_sans_str.strip().split() if san}
-        all_sans = configured_sans | set([_get_public_address()])
-        return sorted(all_sans)
+        extra_sans = set(extra_sans_str.strip().split())
+
+        # Add the ingress addresses of all units
+        extra_sans.add(_get_juju_public_address())
+        binding = self.model.get_binding(CLUSTER_RELATION)
+        try:
+            addresses = binding and binding.network.ingress_addresses
+            if addresses:
+                for addr in addresses:
+                    extra_sans.add(str(addr))
+        except ops.RelationNotFoundError as e:
+            log.error(f"Failed to get ingress addresses for extra SANs: {e}")
```

Comment:
so, the CLUSTER_RELATION should ALWAYS be available b/c it's a peer relation.  it's required to be there 

Would this work?
```suggestion
        binding = self.model.get_binding(CLUSTER_RELATION)
        extra_sans |= {str(addr) for addr in binding and binding.network.ingress_addresses or [])
```

Reply:
This was because of some unit tests that were expecting this relation but it was not there.